torch/_prims/utils.py

```diff
@@ -549,11 +581,14 @@ def check_same_dtype(*args):
             raise RuntimeError(msg)
 
 
-def wrap_scalar(a: Number) -> torch.Tensor:
+def wrap_scalar(a: NumberType, *, dtype: torch.dtype = None) -> torch.Tensor:
     """
     Wraps a Number into a Tensor of corresponding dtype.
     """
-    return torch.tensor(a, dtype=type_to_dtype(type(a)))
+    if dtype is None:
+        return torch.tensor(a, dtype=type_to_dtype(type(a)))
```

Comment:
It's confusing that now type promotion is split between ref (that converts tensor) and here (converts scalar). Maybe wrapping indeed should be done in ref.

Reply:
Yeah I'm going to look at scalar handling more generally and ensure we're preserving scalars properly